### PR TITLE
README for how to install released version

### PR DESCRIPTION
**Pull request by pivotal-nader-ziada**
Fixes: #385 

### Proposed changes
  - readme of how to install using the release yaml
  - link to install.md from repo main README

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Installing the Knative Pipelines component
+
+Before you can run Knative Pipelines, you must install the Knative Pipelines components
+in your Kubernetes cluster. Use this page to add the component to an existing 
+Kubernetes cluster.
+
+## Before you begin
+
+For complete installation instructions of Knative components, see [Installing Knative](https://github.com/knative/docs/blob/master/install/README.md).
```
Comment (bobcatfish) [inline docs/install.md]: would be good to mention here that none of the other Knative components are required for Pipelines to work

(there's a common misconception that istio is required for pipelines for example)

File: docs/install.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+- bucket.service.account.secret.key: the key in the secret with the required
+  service account json. 
+- The bucket is recommended to be configured with a
+  retention policy after which files will be deleted.
+
+Both options provide the same functionality to the pipeline. The choice is based
+on the infrastructure used, for example in some Kubernetes platforms, the
+creation of a persistent volume could be slower than uploading/downloading files
+to a bucket, or if the the cluster is running in multiple zones, the access to
+the persistent volume can fail.
```
Comment (bobcatfish) [inline docs/install.md]: nice!! we should add something here about the entrypoint configuration too, which I think I removed by accident when i re-organized the docs :sweat_smile: https://github.com/knative/build-pipeline/blob/6269b7a216992cf51d370eebc2ca473233049b13/docs/using.md#configure-entrypoint-image

File: docs/install.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+## Adding the Knative Pipelines component
+
+To add only the Knative Pipelines component to an existing installation:
+
+1. Run the
+   [`kubectl apply`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply)
+   command to install [Knative Pipelines](https://github.com/knative/build-pipeline) and its
+   dependencies:
+   ```bash
+   kubectl apply --filename https://storage.googleapis.com/knative-releases/build-pipeline/latest/release.yaml
```
Comment (bobcatfish) [inline docs/install.md]: could we mention something here about installing [versioned releases](https://github.com/knative/build-pipeline/blob/master/hack/release.md#creating-versioned-releases), even if we don't have them yet? or would you rather wait until we have one XD

File: docs/install.md (hunk 1 above)
Comment (pivotal-nader-ziada) [reply docs/install.md]: good point

File: docs/install.md (hunk 2 above)
Comment (pivotal-nader-ziada) [reply docs/install.md]: will add it

File: docs/install.md (hunk 3 above)
Comment (pivotal-nader-ziada) [reply docs/install.md]: I prefer to wait till there is at least one previous release so we don't confuse users, wdyt?

Comment (bobcatfish) [reply docs/install.md]: kk, sounds good!

**Comment by shashwathi** [comment]
There is hold on this PR because there is no knative build pipeline release yet.

File: docs/install.md (hunk 3 above)
Comment (bobcatfish) [reply docs/install.md]: hm actually it looks like we're planning to hold off on merging this until we have a real release (good point @shashwathi XD), so in that case, we could update it pre-emptively with a reference to versioned releases (we'd be starting with 0.1)

Also wondering if we want the path to be `knative-releases/build-pipeline` if we're going to decouple our release from `knative` :thinking:

Comment (bobcatfish) [reply docs/install.md]: + @dlorenc @ImJasonH

File: docs/install.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   time. Use CTRL + C to exit watch mode.
+
+You are now ready to create and run Knative Pipelines: 
+ - See [Knative Pipeline tutorial](./tutorial.md) to get started. 
+ - Look at [the examples](https://github.com/knative/build-pipeline/tree/master/examples)
+
+
+## Configuring Knative Pipelines
+
+### How are resources shared between tasks?
```
Comment (bobcatfish) [inline docs/install.md]: sorry for the late scope creep, but could you mention here also these configurable images: https://github.com/knative/build-pipeline/blob/master/config/controller.yaml#L30 ? ❤️ 😅 

image

File: docs/install.md (hunk 3 above)
Comment (pivotal-nader-ziada) [reply docs/install.md]: I'm not sure I understand what you want to add about installing `versioned releases`, don't we want users to install latest? The link you mention in the comment here is about creating a new versioned release

File: docs/install.md (hunk 4 above)
Comment (pivotal-nader-ziada) [reply docs/install.md]: Changing these images requires editing the release yaml by hand or creating a new custom `release.yaml`, which are advanced cases that I don't think it belong here.

Let me know if you want to add it anyways

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish  fyi, I rebased, which removed the `lgtm`

File: docs/install.md (hunk 4 above)
Comment (bobcatfish) [reply docs/install.md]: > can't say no to this cat :D

hahaha :D

> Changing these images requires editing the release yaml by hand or creating a new custom `release.yaml`, which are advanced cases that I don't think it belong here.

hm interesting! i have to say i do still feel like it's worth documenting - but as a side note i find it weird that we can configure 1 of these images using a configmap and the rest are at runtime. that's another story tho :)

**Comment by bobcatfish** [comment]
If you want to merge as is:

/lgtm

If you want to wait until we have a release and/or add some docs on configuring images via a custom `release.yaml`:

/hold

**Comment by bobcatfish** [comment]
```
I0214 21:53:45.528] --- FAIL: TestTaskRunPipelineRunCancel (306.30s)
I0214 21:53:45.529]     cancel_test.go:165: Error waiting for TaskRun pear-foo-ppsqd to be finish: %!s(<nil>)
I0214 21:53:45.529] FAIL
I0214 21:53:45.538] FAIL	github.com/knative/build-pipeline/test	378.236s
I0214 21:53:45.683] Finished run, return code is 1
```

hm this flake is worrying! as is that format character XD

/test pull-knative-build-pipeline-integration-tests

File: docs/install.md (hunk 3 above)
Comment (bobcatfish) [reply docs/install.md]: > I'm not sure I understand what you want to add about installing `versioned releases`, don't we want users to install latest?

Some users will want latest, but some will want to pin to an explicit version

**Comment by bobcatfish** [comment]
I think what we've got here is a great starting point, let's go for it!

/lgtm
/meow box

**Comment by bobcatfish** [comment]
my bad

/meow boxes

**Comment by bobcatfish** [comment]
Feel free to remove the /hold whenever you're ready @pivotal-nader-ziada ! (or if you want to wait for the release on tuesday)

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish I think its better to merge it in just before you cut a release. Is that okay with you?

**Comment by bobcatfish** [comment]
> @bobcatfish I think its better to merge it in just before you cut a release. Is that okay with you?

Sounds good to me! I think @ImJasonH and I are gonna do that on Tuesday it looks like so we'll take care of this then too.

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish renamed to `Tekton`

**Comment by bobcatfish** [comment]
/lgtm

IT'S HAPPENING :tada::tada::tada:
